uniffi/src/lib.rs

```diff
@@ -297,19 +307,20 @@ unsafe impl ViaFfi for bool {
 /// When serialized in a buffer, strings are represented as a i32 byte length
 /// followed by utf8-encoded bytes. (It's a signed integer because unsigned types are
 /// currently experimental in Kotlin).
-unsafe impl ViaFfi for String {
+unsafe impl FfiConverter for String {
+    type RustType = Self;
     type FfiType = RustBuffer;
 
     // This returns a struct with a raw pointer to the underlying bytes, so it's very
     // important that it consume ownership of the String, which is relinquished to the
     // foreign language code (and can be restored by it passing the pointer back).
-    fn lower(self) -> Self::FfiType {
-        RustBuffer::from_vec(self.into_bytes())
+    fn lower(obj: Self::RustType) -> Self::FfiType {
```

Comment:
`Self::RustType`initially took my head for a spin but after seeing the implementation and understanding all the context behind it I have to say it's a really interesting approach!